Mathlib/Geometry/RingedSpace/OpenImmersion.lean

```diff
@@ -865,6 +865,95 @@ theorem of_stalk_iso {X Y : SheafedSpace C} (f : X ⟶ Y) (hf : OpenEmbedding f.
 
 end OfStalkIso
 
+section
+
+variable {X Y : SheafedSpace C} {f : X ⟶ Y} (H : IsOpenImmersion f)
```

Comment:
The correct thing to do is to make `H` an instance argument and `f` be explicit. The approach in `PresheafedSpace.IsOpenImmersion` is bad.

Reply:
My thought exactly, I will open another PR to address this

Reply:
Can we get #14037 over the line before I open the said PR so that less merge conflict needs to be delt with

Reply:
This approach has been adopted in the latest commit